src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -0,0 +1,277 @@
+                // TODO: user story 104281
+                // to keep previous functionality
+                // unary and binary element-wise operations are supported
+                // will be replaced to snippets opset later
+                const auto op_element_type = op->get_input_source_output(0).get_element_type();
+                if (type_relaxed_node->get_overridden_output_type(0) != op_element_type) {
+                    assert(op->get_output_size() == 1ull);
+
+                    type_relaxed_node->set_overridden_output_type(op_element_type, 0);
+                    std::dynamic_pointer_cast<ngraph::Node>(op)->validate_and_infer_types();
+                }
+            } else {
+                op->validate_and_infer_types();
+            }
+
+            for (auto i = 0ull; i < op->get_output_size(); ++i) {
+                const auto& output = op->output(i);
+                if (output.get_element_type() != op_output_types[i]) {
+                    auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                        output,
+                        op_output_types[i]);
+                    ngraph::copy_runtime_info(output.get_node_shared_ptr(), convert);
+                    for (auto& input : output.get_target_inputs()) {
+                        auto child = input.get_node();
+                        if (child == convert.get()) {
+                            continue;
+                        }
+                        input.replace_source_output(convert->output(0));
+                    }
+                }
+            }
+        }
+    }
+
+    for (auto it = result_types.begin(); it != result_types.end(); ++it) {
+        const auto result = it->first;
+        const auto actual_type = result->get_input_source_output(0).get_element_type();
+        const auto expected_type = it->second;
+        if (actual_type != it->second) {
+            auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                result->get_input_node_shared_ptr(0),
+                expected_type);
+            ngraph::copy_runtime_info(result->get_input_node_shared_ptr(0), convert);
+            result->set_argument(0, convert);
+        }
+    }
+
+    return false;
+}
+
+bool ngraph::snippets::pass::PropagatePrecision::can_be_removed(
+    const element::Type& actual_before,
+    const element::Type& actual_after,
+    const element::Type& required_after) {
+    if (actual_before != required_after) {
+        return false;
+    }
+
+    if ((!actual_before.is_real() || (actual_before == element::bf16)) &&
+        ((actual_after == element::f32) || (actual_after == element::f64))) {
+        return true;
+    }
+
+    return false;
+}
+
+bool ngraph::snippets::pass::PropagatePrecision::can_be_fused(
+    const element::Type& actual_after,
+    const element::Type& required_after) {
+    return
+        (actual_after.is_real() == required_after.is_real()) &&
```

Comment:
What's about case `i8->i32->bf16`? I think we can fused them as well: `i8->bf16`
I think we should rewrite this method as well because `can be fused` depends on` actual_before`:
We can fuse  `i8->i32->bf16` into `i8->bf16` but we cannot  fuse `f32->i32->bf16` into `f32->bf16`

What do you think?

Reply:
i16 & i32 can not be converted to bf16. 
i8 => bf16 and integer & float conversion will be added.

Reply:
done, please, review